stat_var_renaming/stat_var_renaming.py

```diff
@@ -0,0 +1,631 @@
+    prop = row[f"p{constraint}"]
+    if prop in prop_remap:
+      # May need to apply multiple functions for a single property.
+      remapper = prop_remap[prop]
+      for function in remapper:
+        row[f"v{constraint}"] = function(prop,
+                                         row[f"v{constraint}"],
+                                        row['populationType'])
+  return row
+
+
+def generate_dependent_constraint_list():
+  """ Generates a list of dependent variables.
+  Using an OS system call, a protobuf definition is compiled. A definition
+  file is then read in and used to generate a pandas dataframe of depedent
+  variable definitions. 
+
+  Returns: 
+    obs_spec_list -> Observation for statistical variables in 
+        protobuf object format.
+  """
+
+  # Generate population observation spec. Creates a new python file.
+  os.system("protoc -I=. --python_out=. pop_obs_spec_common.proto")
+
+  # Load newly created protobuf class definition.
+  import pop_obs_spec_common_pb2
+  obs_spec_list = pop_obs_spec_common_pb2.PopObsSpecList()
+
+  # Load in PV list from spec proto. Note that covid cases was temporarily
+  # added as a DPV for display, but shouldn't truly be one.
+  with open("pop_obs_spec_nocovid.textproto") as f:
+    counts = f.read()
+    text_format.Parse(counts, obs_spec_list)
+
+  # Create a dataframe that matches the greater stat_vars from DB for merging.
+  dpvs = pd.DataFrame()
+  for spec in obs_spec_list.spec:
+    # Get universal props.
+    new_row = {}
+    new_row['populationType'] = spec.pop_type
+    new_row['measuredProp'] = spec.mprop
+    new_row['statType'] = spec.stat_type
+    
+    # Get independent variables.
+    variables = []
+    for name in spec.cprop:
+      variables.append((name, "", False))
+    # Get depedent variables which depend on the value of the constraint.
+    for name in spec.dpv:
+      variables.append((name.prop, name.val, True))
+    # Variables are sorted alphabetically.
+    variables = sorted(variables)
+
+    # Add as a row to entire dataframe.
+    for index, variable in enumerate(variables):
+      var_name, constraint, is_dpv_var = variable
+      new_row[f"orig_p{index + 1}"] = var_name
+      new_row[f"p{index + 1}_is_dpv"] = is_dpv_var
+      if is_dpv_var:
+        new_row[f"orig_v{index + 1}"] = constraint
+    dpvs = dpvs.append(new_row, ignore_index=True)
+
+  # Only return statistical variables with at least one dependent variable.
+  query_string = ""
+  for c in range(1, MAX_CONSTRAINTS + 1):
+    query_string += f"p{c}_is_dpv == 1 or "
+  return dpvs.query(f"{query_string} False")
+
+
+def remove_depedent_constraints(stat_vars):
+  """ Removes all dependent constraints from list of stat vars.
+  Args: stat_vars: Pandas dataframe holding all stat vars
+
+  Returns: stat_vars with all dependent constraints imputed in place for all 
+    rows.
+  """
+  # Generate list of depedent constraints from protobuf config.
```

Comment:
Please correct the spelling of "dependent", not "depedent", throughout the file.  Thanks.